src/kixi/datastore/metadatastore/command_handler.clj

```diff
@@ -379,6 +379,43 @@ the generated 'update' specs.
                    ::ms/bundled-ids bundled-ids}
                   {:partition-key id}]))))
 
+(defmethod c/command-type->event-types
+  [:kixi.datastore/remove-files-from-bundle "1.0.0"]
+  [_]
+  #{[:kixi.datastore/files-removed-from-bundle "1.0.0"]
+    [:kixi.datastore/files-remove-from-bundle-rejected "1.0.0"]})
+
+(defn reject-remove-files-from-bundle
+  ([cmd reason id bundled-ids]
+   [{::event/type :kixi.datastore/files-remove-from-bundle-rejected
+     ::event/version "1.0.0"
+     :reason reason
+     ::ms/id id
+     ::ms/bundled-ids bundled-ids}
+    {:partition-key id}])
+  ([cmd reason id bundled-ids explain]
+   [{::event/type :kixi.datastore/files-remove-from-bundle-rejected
+     ::event/version "1.0.0"
+     :reason reason
+     ::ms/id id
+     ::ms/bundled-ids bundled-ids
+     :spec-explain explain}
+    {:partition-key id}]))
+
+(defn create-remove-files-from-bundle-handler
```

Comment:
Every time I look at these functions I'm reminded of liberator. not valid is 400, not authorised is 401, not bundle is 404(?). This feels like something we want to abstract out, but I'm not sure what is right yet. (might be taking hints from yada too as @malcolmsparks & @ordnungswidrig are both unhappy with liberator and they both worked on it a lot)